zip_0143.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+from pyblake2 import blake2b
+import struct
+
+from transaction import (
+    MAX_MONEY,
+    OVERWINTER_TX_VERSION,
+    Script,
+    Transaction,
+)
+from tv_output import render_args, render_tv
+from tv_rand import Rand
+
+
+SIGHASH_ALL = 1
+SIGHASH_NONE = 2
+SIGHASH_SINGLE = 3
+SIGHASH_ANYONECANPAY = 0x80
+
+NOT_AN_INPUT = -1 # For portability of the test vectors
```

Comment:
I originally thought this should be ``NOT_AN_INPUT = 0xFFFFFFFF``, but in fact the [C++ code](https://github.com/zcash/zcash/blob/f0daf3915fd81fec7f0f79c4e109fdfaef44d162/src/script/interpreter.h#L23) defines it as ``UINT_MAX`` regardless of the size of ``unsigned int``, so there is no correct cross-platform value to use other than -1. As long as the JSON test vectors use -1, this will work.